mod/consensus/pkg/cometbft/service/genesis.go

```diff
@@ -22,14 +22,26 @@ package cometbft
 
 import (
 	"crypto/sha256"
+	"fmt"
 
 	"github.com/berachain/beacon-kit/mod/consensus-types/pkg/types"
+	"github.com/berachain/beacon-kit/mod/errors"
 	"github.com/berachain/beacon-kit/mod/primitives/pkg/encoding/json"
 	cmtcfg "github.com/cometbft/cometbft/config"
 	"github.com/cometbft/cometbft/node"
 	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
 )
 
+// BeaconGenesisState represents the structure of the
+// beacon module's genesis state.
+//
+//nolint:lll // JSON tags are long.
+type BeaconGenesisState struct {
+	ForkVersion            string                       `json:"fork_version"`
+	Deposits               []types.Deposit              `json:"deposits"`
+	ExecutionPayloadHeader types.ExecutionPayloadHeader `json:"execution_payload_header"`
+}
```

Comment:
q: Don't we write our genesis in BeaconKit as well? If so, we should strive to define genesis structure in a single place and reuse that definition to both write and parse genesis

Reply:
right, noticed our own type of genesis in `consensus-types`. using it, seems more consistent as it is used in other modules like `cli`